From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 29 Jan 2022 22:03:26 +0100
Subject: [PATCH] {BUG in 5.17-rc1} drivers/usb/gadget/udc: show IRQ and not
 ptr in error message

gcc on x86-32 with complains:

    drivers/usb/gadget/udc/at91_udc.c: In function 'at91udc_probe':
    drivers/usb/gadget/udc/at91_udc.h:174:42: error: format '%d' expects
    argument of type 'int', but argument 3 has type 'struct gpio_desc *'
    [-Werror=format=]
      174 | #define DBG(stuff...)           pr_debug("udc: " stuff)
          |                                          ^~~~~~~
    include/linux/printk.h:336:21: note: in definition of macro 'pr_fmt'
      336 | #define pr_fmt(fmt) fmt
          |                     ^~~
    include/linux/dynamic_debug.h:152:9: note: in expansion of macro
    '__dynamic_func_call'
      152 |         __dynamic_func_call(__UNIQUE_ID(ddebug), fmt, func, ##__VA_ARGS__)
          |         ^~~~~~~~~~~~~~~~~~~
    include/linux/dynamic_debug.h:162:9: note: in expansion of macro
    '_dynamic_func_call'
      162 |         _dynamic_func_call(fmt, __dynamic_pr_debug,             \
          |         ^~~~~~~~~~~~~~~~~~
    include/linux/printk.h:570:9: note: in expansion of macro
    'dynamic_pr_debug'
      570 |         dynamic_pr_debug(fmt, ##__VA_ARGS__)
          |         ^~~~~~~~~~~~~~~~
    drivers/usb/gadget/udc/at91_udc.h:174:33: note: in expansion of
    macro 'pr_debug'
      174 | #define DBG(stuff...)           pr_debug("udc: " stuff)
          |                                 ^~~~~~~~
    drivers/usb/gadget/udc/at91_udc.c:1897:33: note: in expansion of
    macro 'DBG'
     1897 |                                 DBG("request vbus irq %d failed\n",
          |                                 ^~~
---
 drivers/usb/gadget/udc/at91_udc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/at91_udc.c b/drivers/usb/gadget/udc/at91_udc.c
index dd0819df096e..9d61b0a26d6c 100644
--- a/drivers/usb/gadget/udc/at91_udc.c
+++ b/drivers/usb/gadget/udc/at91_udc.c
@@ -1895,7 +1895,7 @@ static int at91udc_probe(struct platform_device *pdev)
 					at91_vbus_irq, 0, driver_name, udc);
 			if (retval) {
 				DBG("request vbus irq %d failed\n",
-				    udc->board.vbus_pin);
+				    gpiod_to_irq(udc->board.vbus_pin));
 				goto err_unprepare_iclk;
 			}
 		}
-- 
